src/UI/Panels/ClipboardPanel.cs

```diff
@@ -19,25 +20,76 @@ public class ClipboardPanel : UEPanel
         public override bool ShouldSaveActiveState => true;
         public override bool ShowByDefault => true;
 
-        private static Text CurrentPasteLabel;
+        private static List<GameObject> clipboardRows = new List<GameObject>();
 
         public ClipboardPanel(UIBase owner) : base(owner)
         {
         }
 
+        public static void SelectClipboardItem(int index)
+        {
+            selectedItem = index;
+        }
+
+        /// <summary>
+        /// Adds a new entry and selects it
+        /// </summary>
+        public static void AddNewEntry()
+        {
+            ClipboardItems.Add(null);
+            SelectClipboardItem(ClipboardItems.Count - 1);
+            UpdateCurrentPasteInfo();
+        }
         public static void Copy(object obj)
         {
-            Current = obj;
-            Notification.ShowMessage("Copied!");
+            if (obj == null || ClipboardItems.Count <= selectedItem)
+            {
+                ExplorerCore.Log($"The selected item index is outside the range! {ClipboardItems.Count} is the count.");
+                return;
+            }
+            Copy(obj,selectedItem, true);
+        }
+
+        public static void Copy(object obj, int index, bool showNotification)
```

Comment:
Is the `showNotification` arg necessary?

Reply:
This is just so if you run Copy() from the console it doesn't show the notification. This is mostly useful if you are copying something every second in a coroutine or something. We do want to still use notifications though for when a user clicks the "Copy To Clipboard" button in the inspector.

Reply:
I think it would be good to have a notification if the copied objects, etc. are displayed when copied.
Are there any problems with sending out a large number of notifications in current?